src/tir/op/op.cc

```diff
@@ -112,12 +112,23 @@ void BinaryOpMatchTypes(PrimExpr& lhs, PrimExpr& rhs, Span span) {  // NOLINT(*)
     ICHECK(ltype.lanes() == rtype.lanes()) << "Cannot match type " << ltype << " vs " << rtype;
   }
   if (lhs.dtype() == rhs.dtype()) return;
-  // Only do very simple type coversion
+
+  // We keep casting pretty simple
+  // Two different floating point types will upconvert the lower bit floating point 
+  // to the same type as the higher bit version. E.g. fp16 + fp32 --> fp32 + fp32.
+  // Furthermore:
   // int->float, DataType::Int(32)->int(64)
   // require the types to be relatively consistent
   // This will the reduce amount code generated by operators
   // and also help user to find potential type conversion problems.
```

Comment:
Better to rewire an entire comment here. Here is my thought and feel free to use your own words:
```
  // We keep the dtypes to be relatively consistent to reduce the amount code generated by operators.
  // This can also help users find potential type conversion problems.
  // Specifically, when two dtypes are inconsistent, the int or lower bit one will be upcasted to match
  // another. E.g. fp16 + fp32 --> fp32 + fp32; fp32 + int16 --> fp32 + fp32.
```

Reply:
I've reorganized comments